### PR TITLE
fix copy output command without context

### LINKED ISSUE #201862: VScode “Copy Output” is NOT working when invoked via Command Pallette (Ctrl+Shift+P)
Type: <b>Bug</b>

This is a critical topic for a blind person because one cannot navigate
via shortcuts to a cell’s output (cell) and inspect the generated content.

To repro on VScode (1.85.1) ) (Jan2024) macOS or windows:

use “vscode notebook.ipynb”

1- define python code cell:

some_data = [1,2,3]
print(some_data)

2- execute code cell

get output(cell):  [1,2,3]

3- via command pallete (Ctrl-Shft-P): 

“Notebook: Focus In Active Cell Output”
to get focus to output (cell): [1,2,3]

4- via command pallette (Ctrl-Shft-P):

“Copy Output” generates

VScode alert in Windows and in macOS:
  Command 'Copy Output' resulted in an error
  Cannot use 'in' operator to search for 'outputld' in undefined

Visual Studio Code
Command 'Copy Output'

5- using the “Copy Output” from the […] button on the left to the output (cell) works fine (when using a mouse) to copy the output (cell) content to a scratch pad.

HOWEVER, a blind person cannot navigate to this […] button next to the output (cell) and is unable to review the notebook's output (cell) content.

Bard’s suggestion:
The error message "cannot use 'in' operator to search for 'outputld' in undefined" means that you are trying to use the in operator to check if the property outputld exists in an undefined object. The in operator can only be used to check if a property exists in an object, not in a string, number, or other primitive type.
To fix this error, you need to make sure that the object you are checking is not undefined before you use the in operator. You can do this by using the typeof operator to check if the object is of type object. If it is not, then you can throw an error or simply return false.

￼

VS Code version: Code 1.85.1 (0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2, 2023-12-13T09:48:06.308Z)
OS version: Darwin x64 21.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 x 4000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: unavailable_off<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|8.00GB (0.04GB free)|
|Process Argv|--crash-reporter-id 3817d695-44ba-4cd5-96d5-5b073136718f|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (27)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-perl|cfg|1.19.0
perl-toolbox|d97|1.4.0
dart-code|Dar|3.80.0
flutter|Dar|3.80.0
vscode-eslint|dba|2.4.2
code-runner|for|0.12.1
copilot|Git|1.144.0
copilot-chat|Git|0.11.1
perl-syntax|jac|0.0.7
dart-import|lua|0.3.1
perl-debug|mor|0.6.3
black-formatter|ms-|2023.6.0
isort|ms-|2023.10.1
python|ms-|2023.22.1
vscode-pylance|ms-|2023.12.1
jupyter|ms-|2023.11.1003402403
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
extension-test-runner|ms-|0.0.5
vscode-speech|ms-|0.2.0
dart-syntax-highlighting-only|osc|1.0.1
vscode-print|pdc|0.11.18
quarto|qua|1.110.1
code-spell-checker|str|3.0.1
tabnine-vscode|Tab|3.56.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vshan820:30294714
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
py29gd2263:30899288
vsclangdc:30486549
c4g48928:30535728
azure-dev_surveyone:30548225
f6dab269:30613381
2i9eh265:30646982
0bi6i642:30917235
pythongtdpath:30769146
i26e3531:30792625
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2cf:30928864
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementst:30887150
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
aa_t_chat:30882232
cp7184t3:30927821

```

</details>

### PR DESCRIPTION
**Pull request by amunger**
fix https://github.com/microsoft/vscode/issues/201862

### CODE REVIEW

File: extensions/ipynb/package.json [hunk 1]
```diff
@@ -1,133 +1,138 @@
 {
-	"name": "ipynb",
```
Comment (amunger) [inline extensions/ipynb/package.json:2]: the formatter auto-switched this from tabs to spaces, which matches other package.json files

Comment (amunger) [reply extensions/ipynb/package.json:2]: actually, it's not consistent, but matches with the root level file, and my default formatter settings.

File: extensions/ipynb/package.nls.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
 	"newUntitledIpynb.shortTitle": "Jupyter Notebook",
 	"openIpynbInNotebookEditor.title": "Open IPYNB File In Notebook Editor",
 	"cleanInvalidImageAttachment.title": "Clean Invalid Image Attachment Reference",
-	"copyCellOutput.title": "Copy Output",
+	"copyCellOutput.title": "Copy Cell Output",
```
Comment (rebornix) [inline extensions/ipynb/package.nls.json:9]: As @Tyriar suggested, we could add a Notebook category for the command.

Comment (amunger) [reply extensions/ipynb/package.nls.json:9]: that is done in the commands section of package.json

Comment (amunger) [reply extensions/ipynb/package.nls.json:9]: image

**Review by rebornix — APPROVED** [review]

**Pull request merged**